### PR TITLE
Pull request Sara Carlstein

### PR DESCRIPTION
**Pull request by Sartish**
Link to my deployed project: https://epic-kare-1bb05c.netlify.app/

My reflections on how this week's project turned out:
- Difficult project to structure. But in the end it got a bit easier and a good foundation how to. make changes for next weeks project 

Things I'd like to have clarified or explained in more detail:
Still find states difficult but will probably understand with practise :)

### CODE REVIEW

File: code/src/Form.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import React, {useState} from 'react'
+// import Header from './components/Header/Header'
+import NameForm from './components/NameForm/NameForm'
+import RadioButtons from './components/RadioButtons/RadioButtons'
+import DropDown from './components/DropDown/DropDown'
+import WelcomePage from './components/WelcomePage/WelcomePage'
+import Result from './components/Result/Result'
```
Comment (Ingetora) [inline code/src/Form.js:7]: Nice structuring of code in Components file!

File: code/src/Form.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    //1) prevents the page from refreshing
+    //2) using indexOf method to loop through the array order consisting of different steps
+    //3) Using useStates, so it has a default start" and add +1 to go to next step in array.
+
+    const executeSubmit = (event) => {
+        event.preventDefault();
+        const nextComponentIndex = componentOrder.indexOf(component);
+        setComponent(componentOrder[nextComponentIndex+1])
+        console.log('clicked')
+    }
```
Comment (Ingetora) [inline code/src/Form.js:36]: Easy to follow code thanks to commenting!

File: code/src/Form.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                        {component === 'snackQuestion' && (
+                        <DropDown snack={snack} setSnack={setSnack}/>
+                        )}
+                        {component === 'genreQuestion' && (
+                        <RadioButtons GenreGroup={GenreGroup} setGenreGroup={setGenreGroup}/>
+                        )}
+                        {component === 'nameQuestion' && (
+                        <NameForm name={name} setName={setName}/>
+                        )} 
+                        <button>next</button>
```
Comment (Ingetora) [inline code/src/Form.js:55]: Nice way of creating your form!

File: code/src/components/Result/Result.css [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    width: 300px;
+    margin-top: -5px;
+}
+
+.robot-generator{
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  width: 200px;
```
Comment (Ingetora) [inline code/src/components/Result/Result.css:26]: As I understand, we should use viewport unit as first choice.

**Review by Ingetora — COMMENTED** [review]
Wow, Sara! What a build! Looks GREAT, is fun to try out, and your code is well structured. Fantastic job, wish it was my project! ;o)

File: code/src/Form.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    // This function: 
+    //1) prevents the page from refreshing
+    //2) using indexOf method to loop through the array order consisting of different steps
+    //3) Using useStates, so it has a default start" and add +1 to go to next step in array.
+
+    const executeSubmit = (event) => {
+        event.preventDefault();
+        const nextComponentIndex = componentOrder.indexOf(component);
+        setComponent(componentOrder[nextComponentIndex+1])
+        console.log('clicked')
```
Comment (KarolCislowski) [inline code/src/Form.js:35]: remember to always remove logs 🤓

**Review by KarolCislowski — APPROVED** [review]
Great Project! 
Please remember to always remove logs and commented-out code 🤓 
Maybe for the next step, you will try to add some more exotic types of inputs? 🍹

File: code/src/Form.js (hunk 2 above)
Comment (Sartish) [reply code/src/Form.js:36]: thank you Maria!

File: code/src/Form.js (hunk 5 above)
Comment (Sartish) [reply code/src/Form.js:35]: ahh yes I always tend to forget this, thanks 🤯⭐️

**Pull request closed without merging**